README.md

```diff
@@ -1,37 +1,201 @@
+#### `get(path)`
 
+Get file by path. Return `Promise`;
 
+- `path` (String): the path of the file to get.
 
-## Test
-You may run test with
+```JavaScript
+const foo = await squirrel.get('/foo');
+console.log(foo); // { "foo": "bar" }
+
+const barBaz = await squirrel.get('/bar/baz');
+console.log(barBaz); // { "bar": { "baz": "qux" } }
 ```
-$ npm test
+
+#### `getBy(index, key)`
+
+Get by index value. Return `Promise`;
+
+- `index` (String): the path of the property to get. It need to be declare in [`indexes`'s options](#node-interface)
+- `key` (String): the value to match
+
+```JavaScript
+const foo = await squirrel.getBy('foo', 'bar');
+console.log(foo); // { "foo": "bar" }
+
+const barBaz = await squirrel.getBy('bar.baz', 'qux');
+console.log(barBaz); // { "bar": { "baz": "qux" } }
+```
+
+Field can be nested, as described by [`_.get`](https://lodash.com/docs#get).
+
+#### `getAll(index)`
+
+Get index Map. Return `Promise`;
+
+- `index` (String): the path of the property to get. It need to be declare in [indexes's options](#node-interface)
+
+```JavaScript
+const foo = await squirrel.getAll('foo');
+console.log(foo); // { "bar": { "foo": "bar" } }
+
+const barBaz = await squirrel.getAll('bar.baz');
+console.log(barBaz); // { "qux": { "bar": { "baz": "qux" } } }
 ```
 
-Please note that test use an actual etcd service
+## Command Line Interface
 
+### `squirrel-sync`
+
+Synchronise FS folder with ETCD folder.
+
+```Shell
+$ squirrel-sync --hosts localhost:2379 ./fs-folder /etcd-folder
+```
 
+### `squirrel-watch`
 
+Watch ETCD folder change.
+
+```Shell
+$ squirrel-watch --hosts localhost:2379 /etcd-folder
+```
+
+### Arguments
+
+- `--hosts="host1,host2"`: ETCD hosts. [more](https://github.com/stianeikeland/node-etcd/#etcdhosts--1270012379-options)
+- `--ca=/file.ca`: Ca certificate. [more](https://github.com/stianeikeland/node-etcd/#constructor-options)
+- `--key=/file.key`: Client key. [more](https://github.com/stianeikeland/node-etcd/#constructor-options)
+- `--cert=/file.cert`: Client certificate. [more](https://github.com/stianeikeland/node-etcd/#constructor-options)
+
+# Index System
+
+Squirrel allow to put JSON in file. In this case, it could be indexes to access directly. Consider the following ETCD directory.
+
+```Shell
+/
+├── file1 { "foo": "bar" }
+├── file2 { "foo": "baz" }
+└── file3 { "foo": "qux" }
+```
+
+First at all, we should indicate to squirrel which path we want to index.
+
+```JavaScript
+const squirrel = createSquirrel({
+  indexes: ['foo']
+});
+```
+
+Now, we can get `file1` by its `foo` value.
+
+```JavaScript
+const file1 = await squirrel.getBy('foo', 'bar');
+console.log(file1); // { "foo": "bar" }
+```
+
+We can also get the map value of the index.
+
+```JavaScript
+const fooIndex = await squirrel.getAll('foo');
+console.log(fooIndex);
+/*
+{
+  "bar": { "foo": "bar" },
+  "baz": { "foo": "baz" },
+  "qux": { "foo": "qux" }
+}
+ */
+```
+
+If two files have the same index's value, squirrel keep one of both.
```

Comment:
`the same index value`

Not sure I follow what it does then though. Do you mean `Squirrel keeps one of the two`?